File: src/r/pe_0006.r
# FUNCTIONS


sum_of_squares <- function(vec) {
    sum(vec ^ 2)
}


square_of_sum <- function(vec) {
    sum(vec) ^ 2
}


# SCRIPT


ten <- 1:10
hundred <- 1:100

stopifnot(sum_of_squares(ten) == 385)
stopifnot(square_of_sum(ten) == 3025)
stopifnot(square_of_sum(ten) - sum_of_squares(ten) == 2640)

cat(square_of_sum(hundred) - sum_of_squares(hundred), "\n")

File: src/jl/pe_0006.jl
# FUNCTIONS


function sum_of_squares(vec)
    return sum(n^2 for n in vec)
end

function square_of_sum(vec)
    return sum(vec)^2
end


# SCRIPT


function main()
    vec=1:100
    println(square_of_sum(vec) - sum_of_squares(vec))
end

main()
